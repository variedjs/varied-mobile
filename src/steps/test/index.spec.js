import { mount } from "../../../test";
import VmSteps from "..";
import VmStep from "../../step";

test("icon slot", () => {
  const wrapper = mount(
    {
      template: `
    <vm-steps :active="1">
      <vm-step>
        <template v-slot:inactive-icon>Custim Inactive Icon</template>
        A
      </vm-step>
      <vm-step>
        <template v-slot:active-icon>Custim Active Icon</template>
        B
      </vm-step>
      <vm-step>
        <template v-slot:inactive-icon>Custim Inactive Icon</template>
        C
      </vm-step>
    </vm-steps>
    `
    },
    {
      stubs: {
        // 用来注册自定义组件
        "vm-steps": VmSteps,
        "vm-step": VmStep
      }
    }
  );
  expect(wrapper).toMatchSnapshot();
});

test("click-step event", () => {
  const onClickStep = jest.fn();
  const wrapper = mount(
    {
      template: `
      <vm-steps :active="1" @click-step="onClickStep">
        <vm-step>A</vm-step>
        <vm-step>B</vm-step>
        <vm-step>C</vm-step>
      </vm-steps>
    `,
      methods: {
        onClickStep
      }
    },
    {
      stubs: {
        // 用来注册自定义组件
        "vm-steps": VmSteps,
        "vm-step": VmStep
      }
    }
  );

  wrapper.find(".vm-step").trigger("click");
  expect(onClickStep).toHaveBeenCalledTimes(0);

  wrapper.find(".vm-step-title").trigger("click");
  expect(onClickStep).toHaveBeenCalledWith(0);

  wrapper
    .findAll(".vm-step-circle-container")
    .at(2)
    .trigger("click");
  expect(onClickStep).toHaveBeenCalledTimes(2);
  expect(onClickStep).toHaveBeenLastCalledWith(2);
});

test("inactive-color prop", () => {
  const wrapper = mount(
    {
      template: `
    <vm-steps :active="0" inactive-color="red">
      <vm-step>A</vm-step>
      <vm-step>B</vm-step>
    </vm-steps>
    `
    },
    {
      stubs: {
        // 用来注册自定义组件
        "vm-steps": VmSteps,
        "vm-step": VmStep
      }
    }
  );
  expect(wrapper).toMatchSnapshot();
});
