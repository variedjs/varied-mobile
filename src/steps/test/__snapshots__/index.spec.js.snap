// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon slot 1`] = `
<div class="vm-steps vm-steps-horizontal">
  <div class="vm-steps-items">
    <div class="vm-hairline vm-step vm-step-horizontal vm-step-finish">
      <div class="vm-step-title">
        A
      </div>
      <div class="vm-step-circle-container">Custim Inactive Icon</div>
      <div class="vm-step-line"></div>
    </div>
    <div class="vm-hairline vm-step vm-step-horizontal vm-step-process">
      <div class="vm-step-title vm-step-title-active">
        B
      </div>
      <div class="vm-step-circle-container">Custim Active Icon</div>
      <div class="vm-step-line"></div>
    </div>
    <div class="vm-hairline vm-step vm-step-horizontal">
      <div class="vm-step-title">
        C
      </div>
      <div class="vm-step-circle-container">Custim Inactive Icon</div>
      <div class="vm-step-line"></div>
    </div>
  </div>
</div>
`;

exports[`inactive-color prop 1`] = `
<div class="vm-steps vm-steps-horizontal">
  <div class="vm-steps-items">
    <div class="vm-hairline vm-step vm-step-horizontal vm-step-process">
      <div class="vm-step-title vm-step-title-active">A</div>
      <div class="vm-step-circle-container"><i class="vm-icon vm-icon-check-circle vm-step-icon vm-step-icon-active">
          <!---->
          <!---->
        </i></div>
      <div class="vm-step-line" style="background: red;"></div>
    </div>
    <div class="vm-hairline vm-step vm-step-horizontal">
      <div class="vm-step-title" style="color: red;">B</div>
      <div class="vm-step-circle-container"><i class="vm-step-circle" style="background: red;"></i></div>
      <div class="vm-step-line" style="background: red;"></div>
    </div>
  </div>
</div>
`;
